pkg/apis/core/v1/defaults_test.go

```diff
@@ -1417,6 +1417,42 @@ func TestSetDefaultNamespace(t *testing.T) {
 	}
 }
 
+func TestSetDefaultNamespaceLabels(t *testing.T) {
+	theNs := "default-ns-labels-are-great"
+	s := &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: theNs,
+		},
+	}
+	obj2 := roundTrip(t, runtime.Object(s))
+	s2 := obj2.(*v1.Namespace)
+
+	if s2.Status.Phase != v1.NamespaceActive {
+		t.Errorf("Expected phase %v, got %v", v1.NamespaceActive, s2.Status.Phase)
+	}
+	if _, ok := s2.ObjectMeta.Labels[v1.LabelMetadataName]; !ok {
+		t.Errorf("Missing default namespace label key for %v", s)
+	} else {
+		if s2.ObjectMeta.Labels[v1.LabelMetadataName] != theNs {
+			t.Errorf("Expected default namespace label value of %v, but got %v", theNs, s2.ObjectMeta.Labels[v1.LabelMetadataName])
+		}
+	}
+}
+
+func TestSetDefaultNamespaceLabelsForGenerateName(t *testing.T) {
+	s := &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "default-ns-labels-are-great-generated",
+		},
+	}
+	obj2 := roundTrip(t, runtime.Object(s))
+	s2 := obj2.(*v1.Namespace)
+
+	if _, ok := s2.ObjectMeta.Labels[v1.LabelMetadataName]; !ok {
+		t.Errorf("Missing default namespace label key for %v", s)
+	}
+}
```

Comment:
@liggitt per @jayunit100 comment on previous file, we cannot set default labels on namespaces that uses generated Names, is that right?

If so, probably this test case should be removed?

Reply:
this test is fine, don't remove it... it's covering the case where a name is specified on the namespace.

the integration test covers the generated name scenario (and is currently failing, demonstrating the create path of a namespace with a generated name is not working properly... looks like the namespace strategy Canonicalize method needs to add the same label-setting logic to cover this path)

Reply:
OK, recovered the test, will check the Canonicalize to see what can be done there